src/Materials/Node/nodeMaterialBlock.ts

```diff
@@ -656,18 +656,43 @@ export class NodeMaterialBlock {
         serializationObject.comments = this.comments;
 
         serializationObject.inputs = [];
+        serializationObject.outputs = [];
 
         for (var input of this.inputs) {
             serializationObject.inputs.push(input.serialize());
         }
 
+        for (var output of this.outputs) {
+            serializationObject.outputs.push(output.serialize());

```

Comment:
This one should only serialize the displayName. So maybe adding a boolean to connectionNodePoint.serialize? We do not want to serialize output connection info

Reply:
updated